### ISSUE TITLE
Ordering of variables and legend when using grouped dataframes

### ISSUE THREAD

**Issue by vituri** [JohnCoene/echarts4r#273]
Hi, John!

I'm currently using your newest github version. In some previous cran versions of your package, I was able to choose any order for the elements in the legend. For example, take the iris dataset. Suppose I want that the legend is written in the reverse alphabetical order, that is, starting with Sepal.Width and ending with Petal.Length. Previously, I could just write

```r
library(echarts4r); library(dplyr)

# create a tidy iris and summarise it
df = 
  iris %>% 
  tidyr::pivot_longer(cols = Sepal.Length:Petal.Width,names_to = "name", values_to = "value") %>% 
  group_by(Species, name) %>% 
  summarise(qtd = sum(value)) %>% 
  ungroup()

# arrange by name, so we have the table in the reverse alphabetical order
df %>% 
  arrange(desc(name)) %>% 
  group_by(name) %>% 
  e_charts(Species) %>% 
  e_bar(qtd, stack = 'grp')
```
but now (in the newest version) the legend is always in the alphabetical order. Is it possible to set an arbitrary order in the elements of the legend as before? Maybe this is related to ordering grouped dataframes. I was also previously able to order the data and plot it with e_boxplot, so the result looked like a "staircase", from the biggest to the smallest. This is not possible in the current version. In the following example,

```r
iris %>% 
  arrange(Sepal.Width) %>% 
  group_by(Species) %>% 
  e_charts(reorder = FALSE) %>% 
  e_boxplot(Sepal.Width, reorder = FALSE, outliers = TRUE)
```
I would like that the x axis order be setosa, virginica, versicolor, not the alphabetical order.

I am sure that these things worked in earlier versions because I have several shinyapps running older versions of your package.

**Comment by JohnCoene** [comment]
I would not be surprised it worked in earlier versions. I made some changes to how the groups are treated, I'm confident it's for the best. Essentially, when I created the package there was no function provided by dplyr to split the data by group so I had implemented my own version. I recently made the switch to the more stable `dplyr::group_split`.

I had not noticed until this report that it would cause this. The best way about it is to change the grouped variable to factor and change the levels. Below is a just a basic example but functions from the forcats package as demonstrated in ggplot2 should also work here.

```r
df %>% 
  arrange(desc(name)) %>% 
  mutate(
    name = factor(
      name, 
      c(
        "Sepal.Width",
        "Sepal.Length",
        "Petal.Width",
        "Petal.Length"
      )
    )
  ) %>% 
  group_by(name) %>% 
  e_charts(Species, reorder = FALSE) %>% 
  e_bar(qtd, stack = 'grp')
```
